### PR TITLE
utils cleanup/consistency + moongen

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
* cleanup and consistency
* stage-in moongen as part of sandbox base
* MoonGen folding handling and sync

@seancribbs and co please check my makefiling

### CODE REVIEW

File: docker.mk [hunk 1]
```diff
@@ -16,11 +16,20 @@ DPDK_CONTAINER = dpdk
 CONTAINER ?= $(SANDBOX_CONTAINER)
 BASE_DIR ?= $(SANDBOX_BASE_DIR)
 
+# Our Vagrant setup places MoonGen's repo @ /MoonGen
+# This works off of being relative (../) to utils/sandbox.
+MOONGEN_DIR ?= $(or $(basename $(dirname $(shell pwd)))/MoonGen,\
+~/williamofockham/MoonGen)
+
+FILES_TO_MOUNT := $(foreach f,$(filter-out build libmoon,\
```
Comment (seancribbs) [inline docker.mk]: 👍

**Comment by zeeshanlakhani** [comment]
Utils doesn’t need to know about moongen. Moongen will build on it’s own repo when we make changes. There’s a Travis file there right?

**Comment by zeeshanlakhani** [comment]
Change checking should only care about dpdk and the sandbox itself :).

**Comment by zeeshanlakhani** [comment]
On this repo :). Oh email git lol.

**Comment by drunkirishcoder** [comment]
But doesn’t this PR make sandbox depend on the moongen docker image too? If
moongen gets rebuilt, sandbox will need a refresh too at some point right?

**Comment by zeeshanlakhani** [comment]
@drunkirishcoder yeah, get what you're saying. It's not a biggie right now, but there are a few options, we can go with right away as I finally thought through it. Submodule moongen, which means we force travis to rebuild the sandbox on submodule updates. That's a bit lame. Or, maybe the easiest and best bet is to script the travis.ci for moongen to always build (git clone, build) a new utils image after it deploys. Thoughts? It seems like docker hub also has some webhooks to trigger downstream builds too, e.g.: https://github.com/somleng/dockerhub2ci. Want to pick an option and try it out on your branch? And, maybe I'll merge this first?

**Comment by drunkirishcoder** [comment]
Yeah I already thumbed it. You are good to merge. Just pointing out there’s
some side effect on Travis.

**Comment by zeeshanlakhani** [comment]
Yeah. I think it’d be easy to script upstream changes for moongen -> utils. Whatcha think of those?

**Comment by zeeshanlakhani** [comment]
e.g. @drunkirishcoder -> https://medium.com/mobileforgood/patterns-for-continuous-integration-with-docker-on-travis-ci-71857fff14c5 and there are many others. We can discuss tmrw too.

**Pull request merged**